libraries/chain/db_market.cpp

```diff
@@ -859,38 +867,52 @@ bool database::fill_call_order( const call_order_object& order, const asset& pay
    const asset_object& mia = receives.asset_id(*this);
    FC_ASSERT( mia.is_market_issued() );
 
+   // calculate any margin call fees
+   asset margin_fee(0, pays.asset_id);
+   if (is_margin_call)
+   {
+      margin_fee = calculate_margin_fee( mia, pays );
+      FC_ASSERT( margin_fee.asset_id == pays.asset_id ); // margin fee should be paid in the debt asset
+      // margin fee should never be more than what the call order is receiving (but both could be zero)    
```

Comment:
The fee is not paid from the asset which the call order receiving.

Practically due to BSIP35, `pays` won't be zero.